javascript-bank/bank.js

```diff
@@ -0,0 +1,53 @@
+/* exported Bank */
+function Bank() {
+  this.nextAccountNumber = 1;
+  this.accounts = [];
+}
+
+Bank.prototype.openAccount = function (holder, balance) {
+  if (balance <= 0) {
+    var result = null;
+  } else if (isNaN(balance)) {
+    result = null;
+  } else if (balance > 0) {
+    var account = new Account(this.nextAccountNumber, holder);
+    account.deposit(balance);
+    this.accounts.push(account);
+    result = account.number;
+    this.nextAccountNumber++;
+  }
+  // if (this.accounts.length > 0) {
+  //   this.nextAccountNumber++;
+  // }
+  return result;
+};
+
+Bank.prototype.getAccount = function (number) {
+  var result;
+  if (this.accounts.length > 0) {
+    for (var i = 0; i < this.accounts.length; i++) {
+      if (this.accounts[i].number === number) {
+        result = this.accounts[i];
+      }
+    }
+  } else if (this.accounts.length === 0) {
+    result = null;
+  }
+  return result;
```

Comment:
You can simplify this code a bit. In your `for` loop, `return` the account as soon as it is found. If the `for` loop finishes without `return`ing, then you can `return null`